Homework-2-In-Memory-CRUD-REST-Service/src/routers/userRouter.js

```diff
@@ -0,0 +1,35 @@
+import express from 'express';
+import Joi from 'joi';
+import { createValidator } from 'express-joi-validation';
+import {
+    getAllUsers,
+    getUserById,
+    createUser,
+    updateUser,
+    deleteUser
+} from '../controllers/userController.js';
+
+const schema = Joi.object({
+    login: Joi.string()
+        .alphanum()
+        .trim()
+        .required(),
+    password: Joi.string()
+        .pattern(new RegExp('^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d]{8,}$'))
+        .required(),
+    age: Joi.number()
+        .integer()
+        .min(4)
+        .max(130)
+        .required()
+});
+
+const validator = createValidator({ passError: true });
+
+export const usersRouter = express.Router();
+
+usersRouter.get('/', getAllUsers);
```

Comment:
It's not a part of requirements but what is the point to return protected fields like `password` or serviceFields like `isDeleted`. Also, if it's deleted, should it be still returned?

Reply:
Basically there is no need to return password, cause I don't see usecase when it's needed in the ui layer, but it seems to me that there could be some use cases when some logic will be connected with `isDeleted` property.

According to the task we should make soft delete - not remove the whole item from the collection, but only mark the user as deleted. And I think it should be returned in case of soft delete.

Reply:
Soft deletion is a technic for keeping all the data in db. Even after removal the data is available in db but should be treated as deleted. It could be useful in case if there is some functionality for restoring the data after removal.
And imo it's a bit strange to return soft deleted entries to client that should filter them out.